### DISCUSSION TITLE
Build path on water surface instead of attempting underwater when in free placement mode.

### DISCUSSION THREAD

**Discussion by ocalhoun6** [OpenRCT2/OpenRCT2#17608] [Feature requests]
I'm talking about the mode that lets you place path tiles with a single click, anywhere on the map -- not the mode that allows you to build paths in a continuous track. 

Currently, if you're in this mode and click on water, you get this error:
![Dynamite Dunes 2022-07-25 10-58-56](https://user-images.githubusercontent.com/26573303/180843710-9c0a5262-2961-4442-9a98-c17609489811.png)

Instead of attempting to build underwater and then giving an error, I think it would make more sense to build a path at the water's surface when using this mode.

This could allow players to quickly build paths over water without having to use the other (slower and more tedious) path building mode.

**Comment by HtotheTML** [comment]
If not for the single click path tool, the bridge/tunnel path tool should definitely snap to the water's surface, as presently shift cannot be used to raise/lower path starts.

**Comment by ZehMatt** [comment]
I think the we should just allow to select the height for paths just like with how you can choose the height for flat rides, that is probably the sensible approach and it shouldn't be too much work in theory given the code already exists.

**Reply by awebeer256** [reply]
So that means it'll default to building a "bridge" piece at the water's surface, and you can also hold Shift to move it up or down?

**Reply by HtotheTML** [reply]
Yep, something like that!

**Comment by J-PIE-314** [comment]
The ability to use ctr/shift modifiers in path placement has been added in PR #22569. Does this satisfy your request?

**Reply by ocalhoun6** [reply]
Definitely mitigates the problem, yes ... *but*...

I still don't see any reason why it should default to trying (and failing) to build the path underwater, instead of the default height being the water's surface.

In theory, it should be a pretty simple change to make, right?

**Comment by 733737** [comment]
I agree with the sentiment it should snap to the water's surface but when clearance checks are disabled it would be nice to allow for the shift modifier to build paths underwater in free placement mode since there are builders such as myself that build underwater flat rides and coasters.

**Reply by ocalhoun6** [reply]
Absolutely.

Even when you *don't* have clearance checks disabled, you should be able to use the shift modifier to override the water's surface default and drag it to below the water -- and *that's* when you'd get this error message about not being able to build paths underwater.

**Comment by J-PIE-314** [comment]
Here are ideas of how building underwater will work if auto-building on water surface is implemented:

1. Disable clearance checks requires using the shift modifier to build underwater. (As suggested [above ^](https://github.com/OpenRCT2/OpenRCT2/discussions/17608#discussioncomment-10586250)).
2. Disable clearance checks disables auto-building on water surface (making ease of building on underwater land slopes).
3. Same as idea one, but also implement a visibility toggle (as requested [here](https://github.com/OpenRCT2/OpenRCT2/discussions/18635)) to make water completely invisible, allowing for path placement as if there is no water.
